c_binding/libses.c

```diff
@@ -0,0 +1,731 @@
+/*
+ * Copyright (C) 2016 Red Hat, Inc.
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Author: Gris Ge <fge@redhat.com>
+ */
+
+#define _GNU_SOURCE
+/* ^ For strerror_r() */
+
+#include "libsg.h"
+#include "libses.h"
+#include "utils.h"
+#include "libstoragemgmt/libstoragemgmt_plug_interface.h"
+
+#include <string.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <sys/types.h>
+#include <dirent.h>
+#include <errno.h>
+#include <unistd.h>
+#include <inttypes.h>
+#include <assert.h>
+
+#define _LINUX_SG_DEV_TYPE_SES                  "13"  /* 0x0d  */
+/* ^ SPC-5 Table 139 - PERIPHERAL DEVICE TYPE field */
```

Comment:
doxygen does indeed parse and generate documentation for this file, I just ran it on this branch.